### PR TITLE
Work on withRelationships, finish up frontend role logic, step next will be actually using withRole all over the place

### PR DESCRIPTION
**Pull request by NullVoxPopuli**
The next iteration of roles.
includes:
 - completion if withRole HoC
 - withRelationships HoC for many-to-many models (not currently used anywhere, but this'll be extracted a bit later to react-orbitjs)

### CODE REVIEW

File: source/SIL.AppBuilder.Portal.Frontend/src/data/containers/with-role.tsx [hunk 1]
```diff
@@ -79,24 +79,26 @@ export function withRole<TWrappedProps extends {}>(role: ROLE, givenOptions?: IO
...
 
           resultOfResource = await roleInOrganizationOfResource(currentUser, dataStore, resource, role);
         }
 
         return resultOfOrganization || resultOfResource;
       }
 
-      componentDidMount() {
-        this.doesUserHaveAccess().then(( result ) => {
+      async componentDidMount() {
```
Comment (giancorzo) [inline source/SIL.AppBuilder.Portal.Frontend/src/data/containers/with-role.tsx]: didn't know you could async a lifecycle function. Interesting

Comment (NullVoxPopuli) [reply source/SIL.AppBuilder.Portal.Frontend/src/data/containers/with-role.tsx]: yeah, the return value doesn't matter

File: source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/projects/show/display.tsx [hunk 2]
```diff
@@ -13,6 +13,8 @@ import Owners from './owners';
 import Reviewers from './reviewers';
 import Settings from './settings';
 import Files from './files';
+import { IProvidedProps as ITimeProps } from '@lib/with-moment-timezone';
```
Comment (giancorzo) [inline source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/projects/show/display.tsx]: do you need this?

File: source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/projects/show/index.tsx [hunk 3]
```diff
@@ -21,8 +23,9 @@ export default compose(
   withProjectOperations,
   // todo - extract this to a more general thing?
   withAccessRestriction,
-  /* withRole(ROLE.OrgAdmin, { */
-  /*   redirectTo: '/', */
-  /*   checkOrganizationOf: ({ project }) => project */
-  /* }), */
+  withRole(ROLE.OrganizationAdmin, {
+    redirectTo: '/',
+    checkOrganizationOf: ({ project }) => (project),
```
Comment (giancorzo) [inline source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/projects/show/index.tsx]: parenthesis is not needed `checkOrganizationOf: ({ project }) => project,`

File: source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/projects/show/display.tsx (hunk 2 above)
Comment (NullVoxPopuli) [reply source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/projects/show/display.tsx]: idk, my editor doesn't tell me when I have unused imports.

Comment (NullVoxPopuli) [reply source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/projects/show/display.tsx]: looks like neither of these imports are needed though

**Pull request merged**